packages/routers/src/TabRouter.tsx

```diff
@@ -64,16 +64,14 @@ const getRouteHistory = (
   const history = [{ type: TYPE_ROUTE, key: routes[index].key }];
 
   switch (backBehavior) {
-    case 'initialRoute':
-      if (index !== 0) {
-        history.unshift({ type: TYPE_ROUTE, key: routes[0].key });
-      }
-      break;
     case 'order':
       for (let i = index; i > 0; i--) {
         history.unshift({ type: TYPE_ROUTE, key: routes[i - 1].key });
       }
       break;
+    case 'initialRoute':
+      // The history always begin with the initial route (see changeIndex)
```

Comment:
This is not guaranteed during rehydration. It uses the `getHistory` function to populate the history, so this function shouldn't return incorrect history.